modules/dxp/apps/commerce/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/diagram/DiagramHandler.js

```diff
@@ -0,0 +1,371 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {
+	drag as d3drag,
+	event as d3event,
+	select as d3select,
+	zoom as d3zoom,
+} from 'd3';
+import {openToast} from 'frontend-js-web';
+
+import {
+	DEFAULT_PINS_RADIUS,
+	PINS_CIRCLE_RADIUS,
+	ZOOM_VALUES,
+} from './utilities/constants';
+import {savePin} from './utilities/data';
+import {
+	getAbsolutePositions,
+	getPercentagePositions,
+	isPinMoving,
+} from './utilities/index';
+
+class DiagramHandler {
+	constructor(
+		diagramWrapper,
+		zoomWrapper,
+		imageURL,
+		updateZoomState,
+		setTooltipData
+	) {
+		this._currentScale = 1;
+		this._diagramWrapper = diagramWrapper;
+		this._d3diagramWrapper = d3select(diagramWrapper);
+		this._d3zoomWrapper = d3select(zoomWrapper);
+		this._imageURL = imageURL;
+		this._pinBackground = null;
+		this._setTooltipData = setTooltipData;
+		this._updateZoomState = updateZoomState;
+		this._zoomWrapper = zoomWrapper;
+		this._handleClickOutside = this._handleClickOutside.bind(this);
+		this._handleDragEnded = this._handleDragEnded.bind(this);
+		this._handleDragStarted = this._handleDragStarted.bind(this);
+		this._handleDragging = this._handleDragging.bind(this);
+		this._handleEnter = this._handleEnter.bind(this);
+		this._handleExit = this._handleExit.bind(this);
+		this._handleImageClick = this._handleImageClick.bind(this);
+		this._handleUpdate = this._handleUpdate.bind(this);
+		this._handleZoom = this._handleZoom.bind(this);
+		this._pinsRadius = DEFAULT_PINS_RADIUS;
+
+		this._addListeners();
+		this._printImage();
+		this._addZoom();
```

Comment:
Not sure if these need to be sorted, but personally I'd alphabetize them just for my own personal satisfaction 🤷